app/src/main/java/ru/vssemikoz/newsfeed/data/LocalNewsStorage.java

```diff
@@ -43,7 +44,8 @@ public void updateItem(NewsItem item) {
     }
 
     @Override
-    public void insertUnique(List<NewsItem> newsItems) {
+    public Single<List<NewsItem>> insertUnique(List<NewsItem> newsItems) {
         newsItemDAO.insertUnique(newsItems);
```

Comment:
This is the wrong way. Imagine a factory producing a car. Each part of the conveyor doing some work. All the work is synced across the factory or everything will just break down. You have created the same thing - your Rx-chain is loading news feed from the API. But, this part of the chain will not insert items as expected. It will take it, go out of the factory, save it, return back and give the result. What you should is to use Completable (the example is just an example though):
`Completable.fromAction(new Runnable() {`
`    override void run() {`
`        newsItemDAO.insertUnique(newsItems);`
`    }`
`}).toSingleDefault(newsItems);`